src/extension/src/ProcessHandler.py

```diff
@@ -83,8 +83,8 @@ def start_daemon(self, seq_no, config_settings, ext_env_handler):
             # if process is not running, log stdout and stderr
             if process.poll() is not None:
                 self.logger.log("Process not running for [sequence={0}]".format(seq_no))
-                self.logger.log("Stdout for the inactive process: [Output={0}]".format(str(process.stdout)))
-                self.logger.log("Stderr for the inactive process: [Error={0}]".format(str(process.stderr)))
+                self.logger.log("Stdout for the inactive process: [Output={0}]".format(str(process.stdout.read())))
+                self.logger.log("Stderr for the inactive process: [Error={0}]".format(str(process.stderr.read())))
```

Comment:
I’m pretty much convinced the broken pipe error happens because the extension handler captures the stdout and stderr and later goes away while Core continues to write to it.

See here for someone complaining about this:
https://stackoverflow.com/questions/4106565/newbie-python-subprocess-write-error-broken-pipe

This was never a problem in V1 where what is now Core had the exact same behavior (relating to output/errors) but had a long running equivalent process capture that never went away and constantly kept reading from the IO buffers.

Basically this issue needs to be addressed first in the extension handler and to a lesser degree in Core.

We shouldn’t get a sigpipe if we redirect stdout and stderr to devnull when the Core process is created:

with open(os.devnull, 'r+b', 0) as DEVNULL:
    process = Popen(command, stdin=DEVNULL, stdout=DEVNULL, stderr=STDOUT, close_fds=True)

What this does mean for code in lines 86 and 87 is that IF the process doesn’t start correctly, we can run it again in the mode we run it right now and capture what the failure is.

For Core itself, this means that we can doubly guard against this by comment out line 37 in StdOutFileMirror or try to put exception handling around it (line 37) and ignore any exceptions from writing to the “terminal” which is captured by the dead extension handler process.

Reply:
Will make the changes in the other PR on broken pipe bugfix